README.md

```diff
@@ -1,39 +1,118 @@
 
+```bash
+cd CS340_sample_flask_app
+bash
+virtualenv venv -p $(which python3) 
+source ./venv/bin/activate
+pip3 install --upgrade pip3
 pip install -r requirements.txt
 ```
-Rename the `db_credentials.py.sample` to `db_credentials.py` and put your actual database credentials inside it.
+
+This will create an virtual environment named `venv` and install the required Python packages which are listed inside the file `requirements.txt`.
+
+Next you will need to tweak some files to work with the engineering setup. Because these examples are general there are a few template values that need to be replaced.
+
+Rename the db_credentials.py.sample to db_credentials.py and put your actual database credentials inside it.
 
 
-Using the sample database connectivty webapp
---------------------------------------------
+In particular, inside the current directory `CS340_sample_flask_app` there is a file called `db_credentials.py.sample`. Rename this file to `db_credentials.py` file and put your actual database credentials inside it.
 
+You need to edit the file. It should look like the following code with the curly brackets and their contents replaced with the appropriate content:
+
+```python
+host = 'classmysql.engr.oregonstate.edu'
+user = 'cs340_{your_ONID_username}'
+passwd = '{last-4-digits-of-your-OSU-ID-number}'
+db = 'cs340_{your_ONID_username}'
 ```
-source ./venv/bin/activate
-export FLASK_APP=db_connection_sample.py
-flask run -h 0.0.0.0 -p 5000 --reload
+
+These are the default settings for all students who were enrolled when the term started.  
+
+An **example** might look like this:
+
+```python
+host = 'classmysql.engr.oregonstate.edu'
+user = 'cs340_hedaoos'
+passwd = '1234'
+db = 'cs340_hedaoos'
 ```
 
-The webapp will use the database credentials from db_credentials.py to connect to a database.
+With that file renamed to `db_credentials.py` and the proper credentials added we are almost ready to go.
 
-Now if you open `http://your.server.name/db-test` here, you should see a webpage with results from the database
+
+## Ports and Persistence
+
+Make sure you are still in the `CS340_sample_flask_app` directory and `venv` Python virtual environment. otherwise you should `source ./venv/bin/activate` again. Then setup Flask app file, which is actually a Python script file named `db_connection_sample.py` and run the Flask web server as below:
+
+```bash
+export FLASK_APP=db_connection_sample.py
+flask run -h 0.0.0.0 -p {your_port_number, e.g. 5678} --reload
 ```
 
+Because this is running on a shared machine everyone cannot use port 5000. Everyone will need to use a unique port otherwise you will get an error that the port is in use.
+
+So we specify the Python file for database connection and also set a port number. You could view the database page of nine entries being served by visiting http://access.engr.oregonstate.edu:5678/db-test while you are VPNed into the OSU network.
+
+
+## Forever
```

Comment:
Could you please rename this section to "Running the Flask application persistently" and add a link to:

1) http://flask.pocoo.org/docs/1.0/tutorial/deploy/#run-with-a-production-server
2) http://flask.pocoo.org/docs/1.0/deploying/wsgi-standalone/